src/hyperion/experiment_plans/tests/test_pin_centre_then_xray_centre_plan.py

```diff
@@ -60,3 +68,72 @@ def test_when_pin_centre_xray_centre_called_then_plan_runs_correctly(
 
     mock_detect_and_do_gridscan.assert_called_once()
     mock_pin_tip_centre.assert_called_once()
+
+
+@patch(
+    "hyperion.experiment_plans.pin_centre_then_xray_centre_plan.pin_tip_centre_plan",
+    autospec=True,
+)
+@patch(
+    "hyperion.external_interaction.callbacks.xray_centre.zocalo_callback.XrayCentreZocaloCallback.wait_for_results",
+    lambda self, x: ([0, 0, 0], [1, 1, 1]),
+)
+def test_when_pin_centre_xray_centre_called_then_detector_positioned(
+    mock_pin_tip_centre: MagicMock,
+    test_pin_centre_then_xray_centre_params: PinCentreThenXrayCentreInternalParameters,
+    simple_beamline,
+    test_config_files,
+):
+    sim = RunEngineSimulator()
+    sim.add_handler_for_callback_subscribes()
+    add_simple_pin_tip_centre_handlers(sim)
+    add_simple_oav_mxsc_callback_handlers(sim)
+
+    def add_handlers_to_simulate_detector_motion(msg: Msg):
+        sim.add_handler(
+            "read",
+            "detector_motion_shutter",
+            lambda msg_: {"values": {"value": int(ShutterState.OPEN)}},
+        )
+        sim.add_handler(
+            "read",
+            "detector_motion_z_motor_done_move",
+            lambda msg_: {"values": {"value": 1}},
+        )
+
+    sim.add_wait_handler(
+        add_handlers_to_simulate_detector_motion, "ready_for_data_collection"
+    )
+
+    messages = sim.simulate_plan(
+        pin_tip_centre_then_xray_centre(
+            simple_beamline, test_pin_centre_then_xray_centre_params, test_config_files
+        )
+    )
+
+    messages = messages_from_where(
+        messages, lambda msg: msg.obj is simple_beamline.detector_motion.z
+    )
+    assert messages[0].args[0] == 100
+    assert messages[0].kwargs["group"] == "ready_for_data_collection"
+    assert messages[1].obj is simple_beamline.detector_motion.shutter
+    assert messages[1].args[0] == 1
+    assert messages[1].kwargs["group"] == "ready_for_data_collection"
+    assert (
+        messages := messages_from_where(
+            messages[2:],
+            lambda msg: msg.command == "wait"
+            and msg.kwargs["group"] == "ready_for_data_collection",
+        )
+    )
+    assert messages_from_where(
+        messages[2:],
+        lambda msg: msg.command == "open_run"
+        and msg.kwargs["subplan_name"] == "do_fgs",
+    )
+
+
```

Comment:
Nit: maybe it would be better for this function to live with the run engine simulator? it seems reusable